vividus-engine/src/main/java/org/vividus/steps/VariableResolver.java

```diff
@@ -147,6 +148,10 @@ private Object replaceVariableKeyWithValue(String target, String variableKey)
         {
             return target;
         }
+        if (Optional.of(NULL_VALUE).equals(variableValue))
```

Comment:
this constant is a kind of "internal flag", but there is a chance variable will contain `nullVariable` value and this logic will fail

Reply:
> this constant is a kind of "internal flag", but there is a chance variable will contain `nullVariable` value and this logic will fail
In this case in VariableResolver.java we will get exactly `nullValue `but not `Optional.of("nullValue")`

Reply:
> this constant is a kind of "internal flag", but there is a chance variable will contain `nullVariable` value and this logic will fail

In this case in VariableResolver.java we will get exactly `nullValue `but not `Optional.of("nullValue")`